### DISCUSSION TITLE
How to add specific hidden directories to search paths?

### DISCUSSION THREAD

**Discussion by slafs** [BurntSushi/ripgrep#1588] [General] [answered]
I really like ripgrep. I like ripgrep default rules for ignoring files. In general I think it's a sane default to ignore hidden files and directories. Is there any way to include some hidden directories (or files) to be searched and still preserve the behaviour of not searching in "other" hidden directories/file (i.e. without enabling the `--hidden` option).

I'd like to put some paths in my config file to be always searched, but still skip searching some large hidden directories (e.g. `.git`). This would be useful when one has important stuff in CI directories in the repo (which normally are hidden dirs). E.g.
```
# Always look into those directories:
--glob-add-even-if-hidden-option
.circleci/**
```

I noticed that using `--glob` doesn't work here as it (a) doesn't extend the search paths, but narrows them down to the stuff you put in the value/regex and (b) still needs the `--hidden` option to work.

**Comment by BurntSushi** [comment, marked as answer]
Yup! All you need to do is whitelist the file in an `.ignore` file:

```
$ echo test > foo

$ echo test > .bar

$ rg test
foo
1:test

$ echo '!.bar' > .ignore

$ rg test
foo
1:test

.bar
1:test
```

**Reply by slafs** [reply]
Hmmm, seems to be working for files but not directories? 🤔 
```
$ echo '!.baz/**' >> .ignore

$ mkdir .baz

$ echo test > ./.baz/foo

$ rg test
foo
1:test

.bar
1:test
```

**Reply by BurntSushi** [reply]
I guess `.baz/**` matches everything inside of `.baz` but not `.baz` itself. But you don't need to whitelist everything inside of `.baz`, just `.baz` itself since that's what's hidden:

```
$ echo '!.baz/' >> .ignore

$ rg test
foo
1:test

.bar
1:test

.baz/foo
1:test
```

**Reply by slafs** [reply]
Ah! Perfect! Thanks!
